packages/flutter/lib/src/widgets/app.dart

```diff
@@ -252,6 +264,11 @@ class _WidgetsAppState extends State<WidgetsApp> implements WidgetsBindingObserv
         child: result
       );
     }
+    if (widget.showWidgetInspector || WidgetsApp.showWidgetInspectorOverride) {
```

Comment:
We should probably consider limiting the semantics debugger and widget inspector to debug mode, so that they can get tree-shaken out.